### PR TITLE
man: Implement dynamic manpage titles

### PR DESCRIPTION
**Pull request by chasestruck**
This patch has `man(1)` dynamically set the title of the manpage based
on what section of the manpages you are reading from.

If you are reading from `man1/`, the title is:
    SerenityOS User Manual

If you are reading from `man2/`, `man3/`, or `man4/`, the title is:
    SerenityOS System Programmer's Manual

If you are reading from `man5/`, the title is:
    SerenityOS File Formats Manual

If you are reading from `man6/`, the title is:
   SerenityOS Games Manual

If you are reading from `man7/`, the title is:
   SerenityOS Miscellaneous Information Manual

Finally, if you are reading from `man8/`, the title is:
   SerenityOS System Administration Manual

### CODE REVIEW

**Comment by networkException** [comment]
We already have those name strings in the codebase at various locations, in code [here](https://github.com/SerenityOS/serenity/blob/5caf307ec0c23da13deb106aa7d512bedc77d73e/Userland/Applications/Help/ManualModel.cpp#L13-L22) for example. Maybe instead of duplicating another time that logic could be shared

**Comment by chasestruck** [comment]
So, I avoided something like that here because I wasn't aiming for a 1:1 correlation between these manpage titles and the actual categories. Hence why 3 sections are grouped together under one generic title and some titles use different phrasing. If we want 1:1 correlation I could go ahead and use those strings. Only thing is that if we do want that, wouldn't it be awkward to have `man(1)` depend on `help(1)`? I can't find any other file that has these strings in them.

**Comment by bgianfo** [comment]
> Only thing is that if we do want that, wouldn't it be awkward to have `man(1)` depend on `help(1)`? I can't find any other file that has these strings in them.

You would ideally put it in some shared library (like `LibCore`, or `AK`, A new library? etc..) and then use it from both locations. I agree with networkException, it would be nice to have them synchronized across the codebase.  More consistency, and less debt long term.

**Comment by chasestruck** [comment]
The title names now follow a format of `SerenityOS {category} Manual`, where `{category}` is the category of the manpage you are reading. Thus, a 1:1 correlation between these titles and the manpage categories.

The name strings are still not shared between `man(1)` and `help(1)`. I asked Linus on Discord and he said this was fine for now but that in the future we might want to look into creating a  library to (among other things) share the name strings.

File: Userland/Utilities/man.cpp [hunk 1]
```diff
@@ -116,7 +116,23 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
...
+    else if (*section == '4')
+        title = "SerenityOS Special Files Manual";
+    else if (*section == '5')
+        title = "SerenityOS File Formats Manual";
+    else if (*section == '6')
+        title = "SerenityOS Games Manual";
+    else if (*section == '7')
+        title = "SerenityOS Miscellanea Manual";
+    else if (*section == '8')
+        title = "SerenityOS Sysadmin Tools Manual";
```
Comment (MacDue) [inline Userland/Utilities/man.cpp:135]: I think a slightly neater solution would be to change the `sections` array above to an array of 
```c++
struct Section {
   char const * number;
   char const * name;
}
```
Then here you can just do:  `auto title = String::formatted("SerenityOS {} Manual", section.name)`

Comment (bgianfo) [reply Userland/Utilities/man.cpp:135]: I like that idea!

**Review by MacDue — CHANGES_REQUESTED** [review]
This is quite odd and not what I suggested... Especially the `Section.name[(*section - 48) - 1]`
 
The suggestion was to change the list of sections to an `AK::Array` of structs like so:
(Updating the previous usages of the array accordingly) 
 
```c++

struct Section {
  char const * number;
  char const * section;
};

Array {
  Section { "1", "User Programs" },
  Section { "2", "System Calls" },
  // and so on...
}
```

**Comment by kleinesfilmroellchen** [comment]
Related to #14393

**Comment by chasestruck** [comment]
This patch is now definitely in conflict with that Draft PR posted above, given that most of the changes involve a near-complete rework of the path-finding logic.

File: Userland/Utilities/man.cpp [hunk 2]
```diff
@@ -59,53 +60,68 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
...
+    if (manpage_data.number) {
+        String path = make_path(manpage_data.number);
+        if (access(path.characters(), R_OK) == 0) {
+            for (Section section : sections) {
+                if (*section.number == *manpage_data.number) {
+                    manpage_data.category = section.category;
+                    break;
+                }
+            }
+        }
```
Comment (MacDue) [inline Userland/Utilities/man.cpp:100]: Is this bit really needed? I doesn't look to me like this `access(path)` check was done here before when the manpage number was provided (and you can see it's done again below). 

I think this whole bit could be 
```c++
auto manpage_number = TRY(manpage_data.number.to_int());
manpage_data.category = sections[manpage_number - 1].category;
```

**Review by MacDue — APPROVED** [review]
The changes LGTM :^)

**Comment by kleinesfilmroellchen** [comment]
@chasestruck You can implement this on top of LibManual rather easily, just rebase your changes on #14393 and mark it as draft until that is merged, then you can already make use of LibManual in this PR even before mine is in master. I hope you can figure this out :^)

**Review by kleinesfilmroellchen — APPROVED** [review]

**Pull request closed without merging**